src/lib/connectors/utils/content-type.ts

```diff
@@ -0,0 +1,348 @@
+     *     the same file type (e.g.: for `js` the result will be
+     *     "application/javascript" instead of what this project
+     *     recommends, namely `text/javascript`).
+     *
+     * See also: http://www.iana.org/assignments/media-types/media-types.xhtml
+     */
+
+    /* eslint-disable default-case */
+    switch (fileExtension) {
+        case 'html':
+        case 'htm':
+            return 'text/html';
+        case 'xhtml':
+            return 'application/xhtml+xml';
+        case 'js':
+            return 'text/javascript';
+        case 'css':
+            return 'text/css';
+        case 'ico':
+            return 'image/x-icon';
+        case 'webmanifest':
+            // See: https://w3c.github.io/manifest/#media-type-registration.
+            return 'application/manifest+json';
+        case 'jpeg':
+        case 'jpg':
+            return 'image/jpeg';
+        case 'png':
+            return 'image/png';
+        case 'gif':
+            return 'image/gif';
+        case 'svg':
+            // See: https://www.w3.org/TR/SVG/mimereg.html.
+            return 'image/svg+xml';
+        case 'webp':
+            return 'image/webp';
+        case 'woff2':
+            return 'font/woff2';
+        case 'woff':
+            return 'font/woff';
+        case 'ttf':
+            return 'font/ttf';
+        case 'otf':
+            return 'font/otf';
+    }
+    /* eslint-enable no-default */
+
+    // If the file extension is not in the list above, query `mime-db`.
+
+    return getMediaTypeBasedOnFileExtension(fileExtension);
+};
+
+const determineMediaTypeBasedOnFileType = (rawContent: Buffer): string => {
+    const detectedFileType = fileType(rawContent);
+
+    if (detectedFileType) {
+        // Use the media types from `mime-db`, not `file-type`.
+        return getMediaTypeBasedOnFileExtension(detectedFileType.ext);
+    }
+
+    if (rawContent && isSvg(rawContent)) {
+        // See: https://www.w3.org/TR/SVG/mimereg.html.
+        return 'image/svg+xml';
+    }
+
+    return null;
+};
+
+const parseContentTypeHeader = (response: IResponse): MediaType => {
+    const contentTypeHeaderValue: string = normalizeString(response.headers ? response.headers['content-type'] : null);
+
+    // Check if the `Content-Type` header was sent.
+
+    if (contentTypeHeaderValue === null) {
+        debug(`'content-type' header was not specified`);
+
+        return null;
+    }
+
+    // Check if the `Content-Type` header was sent with a valid value.
+
+    let contentType: MediaType;
+
+    try {
+        if (contentTypeHeaderValue === '') {
+            throw new TypeError('invalid media type');
+        }
+
+        contentType = parse(contentTypeHeaderValue);
+    } catch (e) {
+        debug(`'content-type' header value is invalid (${e.message})`);
+
+        return null;
+    }
+
+    return contentType;
+};
+
+const getContentTypeData = (element: IAsyncHTMLElement, resource: string, response: IResponse) => {
```

Comment:
Add `jsdoc` at least to `getContentTypeData` and `isTextMediaType` (the exported members)